api/turing/validation/validator.go

```diff
@@ -313,58 +314,179 @@ func validateRouterConfig(sl validator.StructLevel) {
 	}
 
 	// Validate traffic rules
-	allRuleRoutesSet := set.New()
 	if router.TrafficRules != nil {
-		if len(router.TrafficRules) > 0 {
-			checkDefaultTrafficRule(sl, "DefaultTrafficRule", router.DefaultTrafficRule)
-			if router.DefaultTrafficRule != nil {
-				allRules := append(router.TrafficRules, &models.TrafficRule{
-					Name:   "default-traffic-rule",
-					Routes: router.DefaultTrafficRule.Routes,
-				})
-				validateDefaultRouteTrafficRules(sl, "TrafficRules", allRules, router.DefaultRouteID)
-				for _, route := range router.DefaultTrafficRule.Routes {
-					allRuleRoutesSet.Insert(route)
-				}
+		validateTrafficRules(sl, instance, router, routeIdsStr)
```

Comment:
Refactored the huge chunk of code into this helper function because of excessive cyclomatic complexity reported by one of our linters.